### ISSUE TITLE
Cache implementation

### ISSUE THREAD

**Issue by ntrp** [square/okhttp#6278]
Sorry my request might come a little harsh but I am kind of pissed.. If there any valid reason why you decided to implement the Cache as a class instead of using an interface thus preventing anybody from implementing his own version of the cache? Let's be honest.. nobody wants to use a non thread safe file system cache in 2020, the era of cloud computing. And just to be clear I am forced to use okhttp as part of a dependency, if it wasn't the case I would not bother annoying you and just pick another http client..

Anyway long story short the feature suggestion is:

**Please rewrite Cache into and interface, copy the code from Cache into FileSystemCache and let the community or user choose if he is willing to implement another caching solution.**

**Comment by swankjesse** [comment]
OkHttp got its start in Android, where an on-disk cache is pretty reasonable. It’s thread-safe BTW, though you can’t use it across multiple processes.

If you’d like to open a new issue that isn’t angry and condescending to OkHttp’s maintainers, please do.
